File: pc4/JsWorker/handlers/downloadHandler.js
const fs = require('fs-extra');
const path = require('path');

const STORAGE_DIR = path.join(__dirname, '../storage');

const handleDownload = (req, res) => {
  const fileName = req.query.file_name;
  if (!fileName) {
    return res.status(400).json({ error: 'No file name specified' });
  }

  const filePath = path.join(STORAGE_DIR, fileName);
  if (!fs.existsSync(filePath)) {
    return res.status(404).json({ error: 'File not found' });
  }

  res.download(filePath);
};

module.exports = { handleDownload };

File: pc4/PyWorker/handlers/download_handler.py
import os
from flask import request, send_from_directory, jsonify

STORAGE_DIR = "storage/"

def handle_download(request):
    file_name = request.args.get('file_name')
    if not file_name:
        return jsonify({"error": "No file name specified"}), 400

    file_path = os.path.join(STORAGE_DIR, file_name)
    if not os.path.exists(file_path):
        return jsonify({"error": "File not found"}), 404

    return send_from_directory(STORAGE_DIR, file_name)